src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -830,7 +830,7 @@ string GenerateCtorParamMetadataInitFunc(TypeGenerationSpec typeGenerationSpec)
                         sb.Append(@$"
     {InfoVarName} = new()
     {{
-        Name = ""{spec.Property.JsonPropertyName ?? spec.Property.ClrName}"",
+        Name = ""{spec.Property.ClrName}"",
```

Comment:
In the cited LOC, would `MemberName ` ever be `null`?

Reply:
It can be for used-defined `JsonPropertyInfo` instances via the contract model. JsonPropertyInfo instances produced by the source generator or the reflection resolver always populate it though.